resources/voting-controls.js

```diff
@@ -0,0 +1,155 @@
+let voted_points = null;
+let voted_note = null;
+let voted_date = null;
+let has_voted = false;
+
+function set_default_vote_data(vote_points, vote_date) {
+    has_voted = true;
+    voted_points = vote_points;
+    voted_date = vote_date;
+    voted_note = JSON.parse(document.getElementById('voted_note').textContent);
+}
+
+let vote_url = '';
+let delete_vote_url = '';
+
+function set_voting_urls(new_vote_url, new_delete_vote_url) {
+    vote_url = new_vote_url;
+    delete_vote_url = new_delete_vote_url;
+}
+
+let translator = null;
+function set_translated_strings(translated_strings) {
+    translator = translated_strings;
+}
+
+let language_code = null;
+function set_language_code(code) {
+    language_code = code;
+}
+
+// JS code constructed from Django templates can contain HTML entities, which aren't escaped when JS
+// modifies the DOM. This function escapes these values, using DOMParser to avoid XSS vulnerability.
+function decodeHTMLString(string) {
+    return new DOMParser().parseFromString(string, 'text/html').documentElement.textContent;
+}
+
+function voteUpdate() {
+    $('#id_has_voted_prompt').show();
+    $('#id_current_vote_value').prop('innerText', voted_points);
+    $('#id_vote_date').prop('innerText', `${translator['last_voted']} ` + voted_date.toLocaleDateString(language_code, {
+        weekday: 'long',
+        year: 'numeric',
+        month: 'long',
+        day: 'numeric',
+    }) + ' at ' + voted_date.toLocaleTimeString(language_code, {
+        hour: 'numeric',
+        minute: 'numeric',
+        hour12: true
+    }));
+    $('#id_has_not_voted_prompt').hide();
+    $('#id_delete_vote_form_submit_button').show();
+
+    $('#id_vote_button').prop('innerText', translator['change_vote'] + ' (' + voted_points + ')');
+    $('#id_points').prop('value', voted_points);
+    $('#id_note').prop('value', voted_note);
+
+    $('#id_points_error_box').hide();
+    $('#id_note_error_box').hide();
+    has_voted = true;
+}
+
+function deleteVoteUpdate() {
+    $('#id_has_voted_prompt').hide();
+    $('#id_has_not_voted_prompt').show();
+    $('#id_delete_vote_form_submit_button').hide();
+
+    $('#id_vote_button').prop('innerText', translator['vote_points']);
+    $('#id_points').prop('value', '');
+    $('#id_points').prop('placeholder', 10);
+    $('#id_note').prop('value', '');
+    $('#id_note').prop('placeholder', decodeHTMLString(translator['note_justification_help_text']));
+
+    $('#id_points_error_box').hide();
+    $('#id_note_error_box').hide();
+    has_voted = false;
+}
+
+function startVotingControls() {
```

Comment:
Let's add some sort of easy way to exist these dialogue boxes. As it stands, users either vote, re-vote, or figure out they can exit by clicking off the box. An `X` in the top of both boxes would be good. Statistics need this too,

Reply:
Should I find a way to have the lightbox class insert this?

Reply:
done